src/actions.py

```diff
@@ -75,17 +75,14 @@ def optimize_limits_action(_action, _target, self):
     graphs.refresh(self)
 
 
-def restore_view_action(_action, _target, self):
-    self.canvas.load_limits()
-    graphs.refresh(self)
-
-
 def view_back_action(_action, _target, self):
-    self.canvas.dummy_toolbar.back()
+    self.canvas.toolbar.back()
+    self.canvas.apply_limits()
 
 
 def view_forward_action(_action, _target, self):
-    self.canvas.dummy_toolbar.forward()
+    self.canvas.toolbar.forward()
+    self.canvas.apply_limits()
```

Comment:
Is the apply_limits() really necessary here? When it sets the view back the previous one, the limits are in principle already set. Seems to work for me when removing this line as well, but I could be missing something or maybe some edge case.